[ML] Update the overview page for machine learning (NNStreamer & NNTrainer) (#1616)
Samsung/tizen-docs: docs/platform/what-is-tizen/overview.md

  ![IoTivity for Connectivity](media/about_tizen_6.png)

## AI and Machine Learning

> ### On-Device & Among-Device AI Pipelines for Inference

Author: Updated!

> NNStreamer (https://github.com/nnstreamer/nnstreamer) is a set of GStreamer plugins that allow GStreamer developers to adopt neural network models easily and efficiently and neural network developers to manage neural network pipelines and their filters easily and efficiently.

Author: Updated!

> - Provide neural network framework connectivities (e.g., tensorflow, caffe) for GStreamer streams.

Author: Updated!

  - Efficient Streaming for AI Projects: Apply efficient and flexible stream pipeline to neural networks.
  - Intelligent Media Filters: Use a neural network model as a media filter / converter.
  - Composite Models: Multiple neural network models in a single stream pipeline instance.

>   - Multi Modal Intelligence: Multiple sources and stream paths for neural network models.

Reviewer: The conventional way to write “Multimodal” should be followed. 
```suggestion
  - Multimodal Intelligence: Multiple sources and stream paths for neural network models.
```

Author: Updated!

- Provide easy methods to construct media streams with neural network models using the de-facto-standard media stream framework, GStreamer.

>   - GStreamer users: use neural network models as if they are yet another media filters.

Reviewer: To maintain consistency with other bullet points, “Users” and “Use” should be capitalized. The word “filter” should be singular here. 
```suggestion
  - GStreamer Users: Use neural network models as if they are yet another media filter.
```

Author: Updated!

>   - Neural network developers: manage media streams easily and efficiently.

Reviewer: ```suggestion
  - Neural Network Developers: Manage media streams easily and efficiently.
```

Author: Updated!

> ### On-Device Training & AI Service Personalization

Reviewer: Please use sentence style capitalization for third headings. Ref: https://github.com/Samsung/tizen-docs/blob/master/styleguide/style.md#capitalization
```suggestion
### On-device training & AI service personalization
```

Author: Updated!

> NNTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on device with user data for the personalization. Using NNtrainer. It is based on the independent open source project NNTrainer(https://github.com/nnstreamer/nntrainer) and provides various applications.

Reviewer: Fixed article usage in the sentence. 
The line _**“Using NNtrainer.”**_ seems incomplete to me here. Should this be a new bullet point? Please look into this. 
```suggestion
NNTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on a device with user data for personalization. Using NNtrainer. It is based on the independent open source project NNTrainer (https://github.com/nnstreamer/nntrainer) and provides various applications.
```

- [NNTrainer: Personalize neural networks on devices!, Samsung Developer Conference 2021](https://www.youtube.com/watch?v=HKKowY78P1A)

> - Features : https://github.com/nnstreamer/nntrainer/blob/main/README.md

Reviewer: ```suggestion
- Features: https://github.com/nnstreamer/nntrainer/blob/main/README.md
```

Author: Updated!

> - Layers

Reviewer: A list should be preceded by a colon.
```suggestion
- Layers:
```

Author: Updated!

>   - Feed-Forword Network : Conv1D, Conv2D, Pooling2D, Flatten, Fully-Connected, etc.

Author: Updated!

>   - Recurrent Layer : LSTM, GRU, RNN, Time Sequencing, etc.

Author: Updated!

  - Pre-processing : reshape, split, translate, etc.

>   - Activation : tanh, sigmoid, relu, softmax

Reviewer: ```suggestion
  - Activation: Tanh, Sigmoid, Relu, Softmax
```

>   - Loss Function : cross entropy (sigmoid, softmax), mse, constant derivative

Reviewer: ```suggestion
  - Loss Function: Cross Entropy (Sigmoid, Softmax), MSE, Constant Derivative
```

Author: Updated!

>   - Optimizer : sgd, adam

Reviewer: ```suggestion
  - Optimizer: SGD, Adam
```

Author: Updated!

>   - Others : weight_initializer, weight_regularizer, weight_decay, learning rate scheduling, teacher forcing, gradient clipping, etc.

Reviewer: ```suggestion
  - Others: Weight_initializer, Weight_regularizer, Weight_decay, Learning Rate Scheduling, Teacher Forcing, Gradient Clipping, etc.
```

Author: Updated!

### On-device & among-device AI pipelines for inference

NNStreamer (https://github.com/nnstreamer/nnstreamer) is a set of GStreamer plugins that allow GStreamer developers to adopt neural network models and also lets neural network developers manage neural network pipelines and their filters easily and efficiently.

- Provide neural network framework connectivities (e.g., TensorFlow, Caffe) for GStreamer streams.
  - Efficient Streaming for AI Projects: Apply efficient and flexible stream pipeline to neural networks.
  - Intelligent Media Filters: Use a neural network model as a media filter / converter.
  - Composite Models: Multiple neural network models in a single stream pipeline instance.
  - Multimodal Intelligence: Multiple sources and stream paths for neural network models.
- Provide easy methods to construct media streams with neural network models using the de-facto-standard media stream framework, GStreamer.
  - GStreamer Users: Use neural network models as if they are yet another media filter.
  - Neural Network Developers: Manage media streams easily and efficiently.

### On-device training & AI service personalization

> NTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on a device with user data for personalization. It is based on the independent open source project NNTrainer (https://github.com/nnstreamer/nntrainer) and provides various applications.

Reviewer 2: Fixed typo.
```suggestion
NNTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on a device with user data for personalization. It is based on the independent open source project NNTrainer (https://github.com/nnstreamer/nntrainer) and provides various applications.
```

> NTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on a device with user data for personalization. Using NNtrainer. It is based on the independent open source project NNTrainer (https://github.com/nnstreamer/nntrainer) and provides various applications.

Reviewer: I think we should remove "Using NNtrainer" because it is an incomplete sentence and seems unnecessary considering the context. Please check this. 

```suggestion
NTrainer is a software framework for training neural network models on embedded devices which have relatively limited resources. Rather than training whole layers of a network from scratch, NNTrainer finetunes the neural network model on a device with user data for personalization. It is based on the independent open source project NNTrainer (https://github.com/nnstreamer/nntrainer) and provides various applications.
```

- [NNTrainer: Personalize neural networks on devices!, Samsung Developer Conference 2021](https://www.youtube.com/watch?v=HKKowY78P1A)
- Features: https://github.com/nnstreamer/nntrainer/blob/main/README.md
- Layers:
  - Feedforward Network: Conv1D, Conv2D, Pooling2D, Flatten, Fully-Connected, etc.
  - Recurrent Layer: LSTM, GRU, RNN, Time Sequencing, etc.
  - Pre-processing: Reshape, Split, Translate, etc.
  - Activation: Tanh, Sigmoid, Relu, Softmax
  - Loss Function: Cross Entropy (Sigmoid, Softmax), MSE, Constant Derivative
  - Optimizer: SGD, Adam
  - Others: Weight_initializer, Weight_regularizer, Weight_decay, Learning Rate Scheduling, Teacher Forcing, Gradient Clipping, etc.
- Applications:
  - MNIST, Reinforcement Learning, Transfer Learning, ResNet, VGG, etc.

### MediaVision
MediaVision provides task based inference API; classification, detection, and pose landmark detection.
It also provides an easy way for a user to do the following:

> - Run models with metafiles which describe model information

Reviewer: The sentence should end with a period. Ref: https://developers.google.com/style/lists#numbered-lettered-bulleted-lists
```suggestion
- Run models with metafiles which describe model information.
```

> - Run various types of models with Machine Learning API

Reviewer: ```suggestion
- Run various types of models with Machine Learning API.
```

> - Use H/W acceleration with NN Runtime

Reviewer: ```suggestion
- Use H/W acceleration with NN Runtime. 
```
